### PR TITLE
feat(angular): v9 with ivy support

### PR DESCRIPTION
**Pull request by NathanWalker**
I took over from where @VladimirAmiorkov left off on [this pr](https://github.com/NativeScript/nativescript-angular/pull/2065) and got things working with Ivy enabled here.

Everyone who sees this does not need to wait until this PR is merged.
You can do the following to try the new v9 by doing the following:

Modify you package.json dependencies as follows:
```
"@nativescript/angular": "https://github.com/nstudio/ns-ng/blob/master/nativescript-angular-9.0.0.tgz?raw=true",
"nativescript-angular": "https://github.com/nstudio/ns-ng/blob/master/nativescript-angular-compat.tgz?raw=true"
```
Then just make sure all your @angular dep's are updated to ~9.1.0, for example:
```
"@angular/core": "~9.1.0"
```

Make you are using at least 1.5.0 of nativescript-dev-webpack in devDependencies.

Add this to your package scripts:

```
"postinstall": "npm run ngcc",
"ngcc": "ngcc --properties es2015 module main --first-only"
```

Then clean you project completely:
```
rm -rf node_modules/ platforms/ package-lock.json hooks
```
Then run with the following:
```
tns run ios --emulator --env.aot --no-hmr
```
Best to always use AoT all the time since Ivy works with it by default and also good to disable hmr as seems there's some other things to get hmr working with ivy.

If you end up seeing an error like this:
```
ERROR in The target entry-point "nativescript-ngx-fonticon" has missing dependencies:
 - file-system
```
You can add a new file `ngcc.config.js` to the root of your project with contents like this which helps the ngcc compiler for Ivy:
```
module.exports = {
  "packages": {
      "@nativescript/angular": {
          ignorableDeepImportMatchers: [
              /tns-core-modules\//,
          ],
          "entryPoints": {
              ".": {
                  "override": {
                      "main": "./index.js",
                      "typings": "./index.d.ts",
                  },
                  "ignoreMissingDependencies": true,
              }
          }
      },
      "nativescript-ngx-fonticon": {
          "entryPoints": {
            ".": {
                "override": {
                  "main": "./nativescript-ngx-fonticon.js",
                  "typings": "./nativescript-ngx-fonticon.d.ts",
                },
                "ignoreMissingDependencies": true,
            }
          }
      },
      "@nota/nativescript-accessibility-ext": {
          "entryPoints": {
            ".": {
                "override": {
                  "main": "./index.js",
                  "typings": "./index.d.ts",
                },
                "ignoreMissingDependencies": true,
            }
          }
      }
  }
}
```
Then try cleaning project and rebuild/run again after adding the ngcc.config.js file.

Here's a sample project configured correctly for Angular 9 with Ivy enabled you can reference (the tns-ios and tns-android runtimes can be 6.4.0 - they are set to specific version just because those are what was installed globally but works fine with 6.4 as well):
https://github.com/vinckobb/nativescript-angular-demo

------------
Lastly if you have a large number of plugins and are encountering too many Ivy/ngcc related issues surrounding various plugins which may not be compatible yet, you can simply set:
Modify your tsconfig.json to include this (parallel to compilerOptions):

```
"compilerOptions": {
  ...your various options, etc.
},
"angularCompilerOptions": {
    "enableIvy": false
}
```

in your tsconfig. **then clean and rebuild.**

That simply disables Ivy and allows you to still build and develop with Angular 9 and then once you have brought all other dependencies up to date with various complier issues, you can then reenable later 👍

### CODE REVIEW

**Comment by NathanWalker** [comment]
@VladimirAmiorkov can you approve again to kick jenkins build? I cleared the lint related issues.

**Comment by rosen-vladimirov** [comment]
test

**Comment by NathanWalker** [comment]
If you find this PR pending and want to use {N} with Angular 9, you can set your `package.json` dependencies to the following:
```
"@nativescript/angular": "https://github.com/nstudio/ns-ng/blob/master/nativescript-angular-9.0.0.tgz?raw=true",
"nativescript-angular": "https://github.com/nstudio/ns-ng/blob/master/nativescript-angular-compat.tgz?raw=true"
```

**Comment by Jonatthu** [comment]
@NathanWalker  Why is this getting too long to get merged? Is there anything more important on nativescript angular team than this?

**Comment by rosen-vladimirov** [comment]
test

**Comment by abcfoundry** [comment]
Any update on this?  @NathanWalker anything we can do to help testing?

**Comment by NathanWalker** [comment]
@abcfoundry I believe it's just clearing the Jenkins e2e build testing which requires a core comment to kick off again. I think those should be clear @rosen-vladimirov - when you get a chance if you can kick off those Jenkins tests again and if you see any remaining issues can you let me know what else is needed to proceed here?

**Comment by abcfoundry** [comment]
@rosen-vladimirov any idea when this will be done?  Are there no resources assigned to Angular?

**Comment by dtopuzov** [comment]
test

**Comment by mavaanan** [comment]
We would also need this for our app (which we already ported to Angular 9). Hope you can progress with this 👍

**Comment by shahraship** [comment]
Looking forward to seeing this merge for completion...

**Comment by kmmccafferty96** [comment]
Any update on this? Thanks!

**Comment by abcfoundry** [comment]
looks like 2 main tests are failing.  with modal and tabs.  @NathanWalker any thoughts?
.......
Modal
Error: Timeout of 600000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/Users/nsbuilduser/workspace/pr-integration-tests-modal-navigation-ng-modal-layout-sim.iPhoneXR.ios13/nativescript-angular/e2e/modal-navigation-ng/e2e/modal-layout.e2e-spec.js)

Tabs...
TypeError: Cannot read property 'statBarHeight' of undefined

**Comment by iameduardod** [comment]
any update @NathanWalker  ?

**Comment by abcfoundry** [comment]
@NathanWalker or @rosen-vladimirov is there anything we can do to push this along?  Ivy is the future for all base Angular projects,  it's very disappointing not have support in NativeScript.  I'm hoping that the Progress Team can realize this and help push this along.  Any thoughts on a timeline would be much appreciated.

**Comment by Arystosedes** [comment]
Any news?

Btw. Is there anybody who uses code sharing and the angular app is working properly with latest angular 9, but for nativescript is used in tsconfig.tns.json:

  "angularCompilerOptions": {
    "enableIvy": false
  },

and it is not working properly? I have an error when trying to compile my app because of:
Execution failed for task ':app:mergeDebugResources'.

Android resource compilation failed
C:\...\platforms\android\app\build\intermediates\incremental\mergeDebugResources\merged.dir\values\values.xml: AAPT: error: file failed to compile.

Is it possible that ns is cannot work even with  "enableIvy": false?

**Comment by abcfoundry** [comment]
@Arystosedes ... I gave up on waiting for NativeScript (Progress) to provide any meaningful response.  This repo has not been updated since Feb.  Rather pathetic.  My guess they are following other channels and really not caring about enterprise and Angular.  

I have moved forward with @NathanWalker and nstudio solution.  This sets you down a path of NX.  I was trying to keep the complexity out of our organization, but given the response from Progress, we dont have a choice.  We are now evaluating React Native.

**Comment by Arystosedes** [comment]
@abcfounry thx for reply. Unfortunately, I have no choice, because I work with my app (after hours) since the end of 2018 and I had the plan to deploy that app to the production environment next month. Even if I can't use Angular 9 with ns, I need to use it with my web app. I'm stuck and wait for any solutions...

**Comment by SmailHammour** [comment]
Any news on this?

**Comment by Meyer-dev** [comment]
Not from the nativescript core team but you can make it work using Nathan
Walker's suggestion. I got it working and it brought my production build
size down by just over 20%. Nativescript animations module I believe still
needs some updates but the workaround there is just to use the angular ones.

**Comment by crowmagnumb** [comment]
I am also going to chime in here with a "any news on this?". I put my angular 9 upgrade on hold when I discovered this issue. I have over a hundred web/mobile shared components that need to be brought up at the same time. So I'm kind of stuck on Angular 8 at the moment until this is resolved. But the silence here from the {N} team is quite worrying. Can someone please respond that you are still staying commited to Angular because I have a GIANT task to figure out alternatives otherwise. I jumped in hard here about a year ago and spent the better part of last year making my core components mobile enabled using Nativescript and building out a mobile app using them. Please tell me that wasn't a complete bust of a year!

**Comment by NathanWalker** [comment]
@crowmagnumb and others inquiring -- due to global pandemic there was some slow down all around but things are slowly picking back up - I can at least assure you that anytime invested with NativeScript and Angular will assuredly be time well spent 👍 For now best to use suggestions in the PR I have provided to begin developing with NativeScript and Angular 9 -- if any issue encountered feel free to share a project and I'd be more than happy to assist further (time permitting however it's easier to assist with a given project and known error stack vs. a shot in dark inquiry).

As far as the PR merging, I need to fix some tests with it and just been a bit overloaded with family and work (no daycare or nanny assistance makes extra free time harder to come by).

We have began updating client apps using the same instructions at the top and can report similar improvements as @Meyer-dev mentioned and there's no danger in updating now. I'd suggest updating and keeping Ivy off if you use a large number of plugins just to begin developing with Angular 9 now. Can always flip it on to try builds with your stack (and plugins) to at least identify plugins that cause you issues and then flip it off to continue developing as normal; Feel free to report here any plugins or errors encountered and we can help look at a few plugin related issues to help community in general there.

Stay safe everyone 🤗

**Comment by kmmccafferty96** [comment]
Thanks for the update @NathanWalker, stay safe as well!

**Comment by abcfoundry** [comment]
@NathanWalker thanks for sharing, but the question of {N} commitment still goes un-answered.  I have tried to reach out to some of the folks at Progress TJ etc..  but still no answer.  

The community really thanks you and your team at nStudio for keeping things moving.

Angular 9 was released on Feb 6 2020 (2 1/2 months ago), and was one of the longest release cycles from the Angular Team.  Yet there has not been any commits to this repo since.  

cheers and stay safe

**Comment by SmailHammour** [comment]
> @NathanWalker thanks for sharing, but the question of {N} commitment still goes un-answered. I have tried to reach out to some of the folks at Progress TJ etc.. but still no answer.
> 
> The community really thanks you and your team at nStudio for keeping things moving.
> 
> Angular 9 was released on Feb 6 2020 (2 1/2 months ago), and was one of the longest release cycles from the Angular Team. Yet there has not been any commits to this repo since.
> 
> cheers and stay safe

I must agree with @abcfoundry. It also came to my attention that the {N} core team at Progress is very unresponsive and even ignoring very important issues. Some reported issues have been open for a year(s) even though the community is begging to have some features/fixes implemented. 

I wish they would be more pro-active and actually tackle down some serious issues in NativeScript.

**Comment by NathanWalker** [comment]
@abcfoundry @SmailHammour totally valid concerns and understandable. I can't speak for the core team as I'm not on the core team but I don't blame them for any nonresponsiveness during this time as well as in the February time frame as much was still undergoing related to Ivy. I know many companies have encountered setbacks in past couple months with the global pandemic.

All I can do is contribute to a framework I myself personally use everyday in many active and ongoing projects. As far as unanswered or pending issue reports (some I have fixed myself in my own day to day operations, of which I plan to contribute PR's when I have the time). In meantime one thing you both could do to help as a fellow community peer of mine is perhaps take 5-10 mins and grab links to the issues from core modules repo and any {N} relayed repo and drop them here. I'd be curious which hit you the hardest, cause you most concern and myself and all those I work with always like to assist the community in ways we can. Look forward to seeing those when you have moment to share 🤗

**Comment by crowmagnumb** [comment]
Yes thank you @NathanWalker, that is fantastic news and all I really needed to hear, that things were still moving along. Sorry for the freak-out. :) Thanks for all your hard work, especially during this time. I have loved working with {N} and look forward to more of the same. Cheers!

**Comment by NidheeshMv** [comment]
Hi @NathanWalker ,

I'm facing below issue while upgrading 

1 export declare class NativeScriptCommonModule {
                       ~~~~~~~~~~~~~~~~~~~~~~~~
node_modules/@nativescript/angular/common.d.ts:1:22 - error NG6002: Appears in the NgModule.imports of DatepickerModule, but could not be resolved to an NgModule class.

This likely means that the library (nativescript-angular/common) which declares NativeScriptCommonModule has not been processed correctly by ngcc, or is not compatible with Angular Ivy. Check if a newer version of the library is available, and update if so. Also consider checking with the library's authors to see if the library is expected to be compatible with Ivy.

I enabled  "importHelpers" : true 
In package.json,
{
script:{
......
postinstall : "ngcc"
}
}

could you please help me on this

**Comment by NathanWalker** [comment]
@NidheeshMv possible to share entire package.json and tsconfig.json? Sometimes the most subtle thing can cause a compile issue —- lastly most helpful if I’d be able to run your project as is (can be private share or is public) —- could probably look late tomorrow or Monday.

**Comment by d-koppenhagen** [comment]
Hey, if there is anything I can do to help (contributing, testing, etc.), just let me know. As I am really interested to get things working with Angular9 and also with the schematics for it (https://github.com/NativeScript/nativescript-schematics/issues/275), I will do my best to support you.

**Comment by NathanWalker** [comment]
Thanks @d-koppenhagen I just emailed you 🤗

**Comment by ElecTreeFrying** [comment]
+1

**Comment by SmailHammour** [comment]
> @abcfoundry @SmailHammour totally valid concerns and understandable. I can't speak for the core team as I'm not on the core team but I don't blame them for any nonresponsiveness during this time as well as in the February time frame as much was still undergoing related to Ivy. I know many companies have encountered setbacks in past couple months with the global pandemic.
> 
> All I can do is contribute to a framework I myself personally use everyday in many active and ongoing projects. As far as unanswered or pending issue reports (some I have fixed myself in my own day to day operations, of which I plan to contribute PR's when I have the time). In meantime one thing you both could do to help as a fellow community peer of mine is perhaps take 5-10 mins and grab links to the issues from core modules repo and any {N} related repo and drop them here. I'd be curious which hit you the hardest, cause you most concern and myself and all those I work with always like to assist the community in ways we can. Look forward to seeing those when you have moment to share 🤗

@NathanWalker This one for example has been open since 2018 -> https://github.com/NativeScript/nativescript-angular/issues/1367

**Comment by cdmistman** [comment]
@NathanWalker This one hasn't been open for long but I'm curious if you have a patch ready https://github.com/NativeScript/NativeScript/issues/8551

It was closed because the original wording of the issue was due to an issue in my own code, however I discovered another bug that's related.

**Comment by NathanWalker** [comment]
@SmailHammour thanks! yep definitely familiar with that one :) - In short there's a lot of angular router paradigms that are well suited for web scenarios which don't necessarily present 1-1 in mobile scenarios and believe a lot of issues like that can often be solved by taking a good look at the overall mobile route setup and often simplifying the approach. Additionally I've been considering writing an alternate angular router for {N} which would avoid a lot of the trouble that issue represents. It's definitely an ongoing debate however the Vue integration took a similar approach where integrating the vue-router didn't make sense 1-1 and instead introduced a much simpler and vastly more stable Navigation service to handle mobile scenarios better with vastly better results.

@cdmistman Have not seen that one - thanks for raising it here; I'll be interested to dig into that one in the coming weeks. If you have an immediate blocker due to that you may could email team@nstudio.io to see if more immediate assistance could be possible.

**Comment by crowmagnumb** [comment]
So I created a repo similar to the one @NathanWalker posted above for others who want to dabble in the move to Angular 9/Ivy. It's located [here](https://github.com/animalus/testns-app). In the master branch I was able to successfully make a simple app to display a dialog. But in the side-drawer branch I am trying to use the `nativescript-ui-sidedrawer` plugin and I get the "missing dependencies" problem but can't get the work-arounds to work. If anyone has any ideas on how to fix it I would love that. Happy to reciprocate with counter-help and I offer this repo up as a starting point for anyone of course.

**Comment by kmmccafferty96** [comment]
@crowmagnumb I was seeing that too, this fixed it for me https://github.com/NativeScript/nativescript-ui-feedback/issues/1355

**Comment by crowmagnumb** [comment]
@kmmccafferty96 Thanks for that! That gave me hope but then I ran into other issues that are documented in my project [README](https://github.com/animalus/testns-app) and a little too wordy to recreate here. It boils down to the fact that the `nativescript-ui-sidedrawer` is not Ivy compatible, but disabling Ivy I still can't complie my simple little tiny project.  :(

**Comment by Meyer-dev** [comment]
I have 3 projects with sidedrawer working with ivy. Just keep trying :) did
you follow Nathan's recommendations regarding the ngcc file? That did the
trick for me.

**Comment by crowmagnumb** [comment]
@Meyer-dev That's encouraging! Thank you. I do indeed have a `ngcc.config.js` in the root of my directory in the `side-drawer` branch. It might be set to DISABLED by the time you get there because I'm experimenting but its contents are and it doesn't seem to do anything. This is automatically seen by ngcc right, I don't have to set anything else? What do yours look like? Cheers.
```
module.exports = {
    packages: {
        "@nativescript/angular": {
            entryPoints: {
                ".": {
                    override: {
                        main: "./index.js",
                        typings: "./index.d.ts"
                    },
                    ignoreMissingDependencies: true
                }
            }
        },
        "side-drawer-directives": {
            entryPoints: {
                ".": {
                    override: {
                        main: "./side-drawer-directives.js",
                        typings: "./side-drawer-directives.d.ts"
                    },
                    ignoreMissingDependencies: true
                }
            }
        }
    }
};
```

**Comment by Meyer-dev** [comment]
here is me ngcc.config.ts file which is slightly different and working.Let
me know if this helps.

module.exports = {
    "packages": {
        "@nativescript/angular": {
            "entryPoints": {
                ".": {
                    "override": {
                        "main": "./index.js",
                        "typings": "./index.d.ts",
                    },
                    "ignoreMissingDependencies": true,
                }
            }
        },
        /*
        "dataform-directives": {
            "entryPoints": {
                ".": {
                    "override": {
                        "main": "./dataform-directives.js",
                        "typings": "./dataform-directives.d.ts",
                    },
                    "ignoreMissingDependencies": true,
                },
                "angular": {
                    "override": {
                      "main": "./dataform-directives.js",
                      "typings": "./dataform-directives.d.ts",
                    },
                    "ignoreMissingDependencies": true,
                  }
            }
        },
        */
        "nativescript-ui-dataform": {
          "entryPoints": {
              "angular": {
                  "override": {
                    "main": "./dataform-directives.js",
                    "typings": "./dataform-directives.d.ts",
                  },
                  "ignoreMissingDependencies": true,
                }
          }
      },

        "nativescript-ui-sidedrawer": {
            "entryPoints": {
              ".": {
                "override": {
                  "main": "./ui-sidedrawer.ios.js",
                  "typings": "./index.d.ts",
                },
                "ignoreMissingDependencies": true,
              },
              "angular": {
                "override": {
                  "main": "./side-drawer-directives.js",
                  "typings": "./side-drawer-directives.d.ts",
                },
                "ignoreMissingDependencies": true,
              }
            }
          },

          "nativescript-ui-listview": {
            "entryPoints": {
              ".": {
                "override": {
                  "main": "./ui-listview.ios.js",
                  "typings": "./index.d.ts",
                },
                "ignoreMissingDependencies": true,
              },
              "angular": {
                "override": {
                  "main": "./listview-directives.js",
                  "typings": "./listview-directives.d.ts",
                },
                "ignoreMissingDependencies": true,
              }
            }
          },




        "nativescript-ngx-fonticon": {
            "entryPoints": {
              ".": {
                  "override": {
                    "main": "./nativescript-ngx-fonticon.js",
                    "typings": "./nativescript-ngx-fonticon.d.ts",
                  },
                  "ignoreMissingDependencies": true,
              }
            }
        },
        "@nota/nativescript-accessibility-ext": {
            "entryPoints": {
              ".": {
                  "override": {
                    "main": "./index.js",
                    "typings": "./index.d.ts",
                  },
                  "ignoreMissingDependencies": true,
              }
            }
        },

    }
  }

**Comment by crowmagnumb** [comment]
I'll say up-front, because maybe we are annoying others  :), if anyone would like to start a small group of some kind that we can bounce ideas and little gotchas that we all experience off each other please let me know. I would love that. We would all like to just get to coding and not dealing with setup.  :) Email me at my github username at gmail.

@Meyer-dev I have made a modification based on yours mine now reads like that below. But what I get with this gcc config set is ...

```
ERROR in node_modules/nativescript-ui-sidedrawer/angular/side-drawer-directives.d.ts:79:22 - error NG6002: Appears in the NgModule.imports of AppModule, but could not be resolved to an NgModule class.

This likely means that the library (nativescript-ui-sidedrawer/angular) which declares NativeScriptUISideDrawerModule has not been processed correctly by ngcc, or is not compatible with Angular Ivy. Check if a newer version of the library is available, and update if so. Also consider checking with the library's authors to see if the library is expected to be compatible with Ivy.

79 export declare class NativeScriptUISideDrawerModule {
```

gcc.config.js:

```
module.exports = {
    packages: {
        "@nativescript/angular": {
            entryPoints: {
                ".": {
                    override: {
                        main: "./index.js",
                        typings: "./index.d.ts"
                    },
                    ignoreMissingDependencies: true
                }
            }
        },
        "nativescript-ui-sidedrawer": {
            entryPoints: {
                ".": {
                    override: {
                        main: "./ui-sidedrawer.android.js",
                        typings: "./index.d.ts"
                    },
                    ignoreMissingDependencies: true
                },
                angular: {
                    override: {
                        main: "./side-drawer-directives.js",
                        typings: "./side-drawer-directives.d.ts"
                    },
                    ignoreMissingDependencies: true
                }
            }
        }
    }
};
```

**Comment by crowmagnumb** [comment]
So, I switched over to my MAC to run it in the ios simulator and it works! Without any ngcc.config.js file even! So its just running android (I run under Ubuntu) that I'm getting all my issues. :( Any thoughts on that?

**Comment by crowmagnumb** [comment]
UPDATE: If I **don't** set the `--env.aot` flag it works fine. I just need to run `tns run android` and then it works (also works for me on mac/ios this way too). If I add the `--env.aot` flag with android I get the error I posted above.

**Comment by iameduardod** [comment]
An alternative is switching to a framework that already supports v9 https://twitter.com/LiamDeBeasi/status/1261298258506375169

**Comment by crowmagnumb** [comment]
@iameduardod I should update here that I am now having success with the above "workaround" and would gladly help others with their transition to it as well with some of the knowledge that I've gained from this exercise. And I have intel that the workaround is very temporary and will be made permanent in a short while.

**Comment by mapo80** [comment]
> @iameduardod I should update here that I am now having success with the above "workaround" and would gladly help others with their transition to it as well with some of the knowledge that I've gained from this exercise. And I have intel that the workaround is very temporary and will be made permanent in a short while.

A migration case would be very helpful and what steps are to be followed.

**Comment by thegnuu** [comment]
I was able to get my application working (partially) with this workaround, but we are using a custom dynamic component loader from an external url in our electron application and I want to rebuild this feature for our new NativeScript application, everything works until I want to build the angular module within our application.

If I try to use `Compiler` from `@angular/core` I receive the following error: `Error: Runtime compiler is not loaded`.

Has anyone an idea how to fix that?

**Comment by thegnuu** [comment]
Never mind, I found a solution for my issue, I just had to enable Ivy in my case... I did not double check this while copying the files, since my externals modules are built with Ivy enabled the app it self of course has to be built with Ivy as well.

**Comment by NidheeshMv** [comment]
@NathanWalker 
 I'm getting below error while tns run ios . All dependency packages upgraded to Angular 9

 '** ERROR in The given rootPath undefined is not a file of the program **.'

Could you provide any solution to fix this

**Comment by samirotiv** [comment]
@NathanWalker Much appreciate your effort. 

It will be great if you could explain, what is currently between us and getting this PR merged and finished?

Asking because I will be happy to be involved in contributing.

**Comment by NathanWalker** [comment]
This PR will be merged and published next week no later than Wednesday 👍

**Comment by NathanWalker** [comment]
Published in `@nativescript/angular`: `~9.0.0` now. Please [see here for update instructions and tips](https://github.com/NativeScript/nativescript-angular/wiki/Updating-and-developing-for-@nativescript-angular-v9-with-Ivy...-or-without.)

**Comment by samirotiv** [comment]
Great work! :)

**Comment by westlakem** [comment]
This still doesn't work on a brand new project

**Comment by NathanWalker** [comment]
@westlakem could you share the exact commands you’re using to create?

**Comment by westlakem** [comment]
I followed the instructions on 
https://docs.nativescript.org/angular/code-sharing/creating-a-new-project.html to create a shared project with SCSS. The command I ran was `ng new -c=@nativescript/schematics sass-project --shared --style=scss`. I attempted to fix it following some other articles, with little success.  You can see my journey in stack overflow. https://stackoverflow.com/questions/62921701/unable-to-create-angular-native-shared-application/62924265#62924265

**Comment by kukurt123** [comment]
Everything works fine until i installed a plugin with element-registry. i think it is the nativescript-ui-dataform. my error goes like this ::::: 
ERROR in The target entry-point "undefined" has missing dependencies:
 - nativescript-angular/element-registry

i followed all the instruction and placed the recommended code inside ngcc.config.js still that dang error exists. -_-

**Comment by NathanWalker** [comment]
@kukurt123 we’d recommend updating to Angular 10 and using UI-dataform 
7.0.0 latest version as it’s ivy compatible with 10 and doesn’t need to be in ngcc.config if you just update to angular 10.
